zerver/views/registration.py

```diff
@@ -465,10 +468,28 @@ def accounts_register(request: HttpRequest) -> HttpResponse:
             "MAX_NAME_LENGTH": str(UserProfile.MAX_NAME_LENGTH),
             "MAX_PASSWORD_LENGTH": str(form.MAX_PASSWORD_LENGTH),
             "MAX_REALM_SUBDOMAIN_LENGTH": str(Realm.MAX_REALM_SUBDOMAIN_LENGTH),
+            "EMAIL_ADDRESS_VISIBILITY_EVERYONE": email_address_visibility,
+            "EMAIL_ADDRESS_VISIBILITY_ADMINS": email_address_visibility,
+            "EMAIL_ADDRESS_VISIBILITY_NOBODY": email_address_visibility,
+            "EMAIL_ADDRESS_VISIBILITY": email_address_visibility,
```

Comment:
You don't need to send 4 context variables here. Sending just one (`"EMAIL_ADDRESS_VISIBILITY"`) should be sufficient.

Reply:
I believe those contexts are set along with the realm settings, so one would not be sufficient given there are 3 possible settings/displays.